File: Node/studentapi/studentapi/src/main/java/com/adobe/studentapi/model/Student.java
package com.adobe.studentapi.model;

import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

@Document("students")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Student {
    @Id
    private String id;

    @Indexed(unique = true)
    private String uid;

    private String name;
    private String phone;
    private String fatherName;
    private String motherName;
}

File: Node/models/Student.js
const mongoose = require("mongoose");

// Student model
const studentSchema = new mongoose.Schema({
  uid: { type: String, required: true, unique: true },  // make uid unique here
  name: { type: String, required: true },
  phone: String,
  fatherName: String,
  motherName: String,
});

const Student = mongoose.models.Student || mongoose.model("Student", studentSchema);

module.exports = Student;
